File: app/models/course.rb
class Course < ActiveRecord::Base
  belongs_to :candidate

  validates :candidate_id, presence: true
  validates :title , presence: true
  validates :description , presence: true
  validates :semester , presence: true
  validates :professor, presence: true
  validates :department , presence: true

  scope :visible, -> { where(is_visible: true) }
end

File: app/assets/javascripts/app/features/courses/models/course.js
'use strict';

angular.module('bridge')

  .run(['modelize', function (modelize) {
    modelize.defineModel('course', {
      baseUrl: '/candidate/courses',

      candidateId: null,
      educationRecordId: null,
      title: '',
      description: '',
      semester: '',
      professor: '',
      department: '',
      isVisible: true
    });
  }])

  .factory('Course', ['modelize', function (modelize) {
    return modelize('course').$modelClass;
  }]);
